### PR TITLE
GamePicker now picks the same game each day for each set of players

### PR DESCRIPTION
**Pull request by Serneum**
TODO: Write tests

### CODE REVIEW

File: lib/nermesterts/game_picker.ex [hunk 1]
```diff
@@ -26,6 +26,8 @@ defmodule GamePicker do
       nil
   """
   def pick_game(games_list, num_players) do
+    :rand.seed(:exsplus, DateTime.utc_now |> DateTime.to_date |> Date.to_erl)
```
Comment (daveshah) [inline lib/nermesterts/game_picker.ex]: So, this and the call to `Enum.take_random` are non-deterministic...
That said, if you want to test this and allow for this to be deterministic, maybe a "game randomization" function would be in order - something that you could inject.

There's a number of ways to inject it, but what this *could* look like:
```elixir
def pick_game(games_list, num_players, randomizer \\ GamePicker.Randomizer) do 
...
|> randomizer.randomize()
end
```
You could then test the injected piece independently if you want. 
Just a thought.

File: lib/nermesterts/game_picker.ex [hunk 2]
```diff
@@ -26,6 +26,8 @@ defmodule GamePicker do
...
+
```
Comment (mbramson) [inline lib/nermesterts/game_picker.ex]: Ha! I love this solution. It's hacky in a really elegant way.

File: lib/nermesterts/game_picker.ex (hunk 1 above)
Comment (mbramson) [inline lib/nermesterts/game_picker.ex]: Does Enum.take_random not use the value seeded to :rand.seed/2?

Comment (Serneum) [inline lib/nermesterts/game_picker.ex]: It does

Comment (Serneum) [inline lib/nermesterts/game_picker.ex]: Looking at something like this:

```elixir
defmodule Randomizer do
  def new do
    :rand.seed(:exsplus, DateTime.utc_now |> DateTime.to_date |> Date.to_erl)
  end

  def new(seed) do
    :rand.seed(:exsplus, seed)
  end
end
```

Is that how the randomizer would work with injection?

Comment (lojack) [inline lib/nermesterts/game_picker.ex]: @daveshah I'd like to respectively disagree that this is non-deterministic. Given the same seed and the same pseudorandom algorithm, you should always get the same output. The seed is what gives the RNG its non-determinism (which we're taking away)

Comment (lojack) [inline lib/nermesterts/game_picker.ex]: But, I do agree that this randomizer should be extracted and isolated.

Comment (Serneum) [inline lib/nermesterts/game_picker.ex]:
```elixir
defmodule GamePicker.Randomizer do
  def new do
    :rand.seed(:exsplus, DateTime.utc_now |> DateTime.to_date |> Date.to_erl)
  end

  def new(seed) do
    :rand.seed(:exsplus, seed)
  end

  def randomize(games_list) do
    games_list
    |> Enum.take_random(1)
    |> List.first
  end
end
```

**Comment by Serneum** [comment]
Tests will currently fail

File: test/nermesterts/game_picker_test.exs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  use ExUnit.Case
+
+  alias Nermesterts.Game
+  alias GamePicker
+
+  test "game selection is consistent when the day and number of players are the same" do
+    games = [%Game{name: "Game1", min_players: 2, max_players: 4},
+             %Game{name: "Game2", min_players: 2, max_players: 4},
+             %Game{name: "Game3", min_players: 2, max_players: 4}]
+    randomizer = GamePicker.Randomizer.new({2016, 10, 17})
```
Comment (daveshah) [inline test/nermesterts/game_picker_test.exs]: This won't work because `new` returns the result of the last called function. Not the `Randomizer`.

**Comment by Serneum** [comment]
Yeah, Eric had narrowed that down. I was still treating it like it was OO

**Comment by ericworkman** [comment]
Todo

- [ ] Rename randomizer
- [ ] Make the tests great again
- [ ] Refactor randomizer cause it ain't too pretty (and namespace it better)

In short, this randomizer uses the current year and day of year to as the index of the list of available games to reliably choose the same game for a given day and game list. I'm not happy with the tests, as I'm not too sure about how to set this up to be tested reliably (but I barrelled through with it anyway). Paired with @Serneum while doing this.

**Pull request merged**